### DISCUSSION TITLE
Responsive styling

### DISCUSSION THREAD

**Discussion by JClackett** [mantinedev/mantine#2422] [Feature requests]
Any chance of getting a syntax similar to chakra for responsive styling?

```tsx
<Text fontSize={{ base: 'sm', md: 'md', lg: 'lg' }}>
  This is responsive text
</Text>
```

It's just so easy to do responsive styling this way.

If not, what is the reason?

I think this is the only thing holding me back from using Mantine at the moment

**Comment by JClackett** [comment]
Comparing the 'verbose'ness of this:


```tsx
// have to make a random style object and use the theme object
const useStyles = createStyles((theme) => ({
  container: {
    fontSize: theme.fontSizes["sm"],
    [`@media (max-width: ${theme.breakpoints.lg}px)`]: {
      fontSize: theme.fontSizes["lg"]
    },
    [`@media (max-width: ${theme.breakpoints.xl}px)`]: {
      fontSize: theme.fontSizes["xl"]
    },
  },
}));

// then need to use a hook
const { classes } = useStyles();

// then apply styles

<Text className={classes}>Some text</Text>

```


to:

```tsx
<Text fontSize={{ base: 'sm', lg: 'lg', xl: 'xl' }}>
  This is responsive text
</Text>
```

**Comment by rtivital** [comment]
It is planned to be added as part of 6.0 release

**Reply by JClackett** [reply]
That is incredible news 🎉

**Reply by ylluminate** [reply]
Did this ever make it in so as to allow:
https://chakra-ui.com/docs/styled-system/responsive-styles#the-array-syntax
![example](https://p153.p0.n0.cdn.zight.com/items/jku8XbY2/5f4eefc5-b7d3-4291-9e67-0d50e26b742f.png?v=49045047bb3e76fdea3f39618d3ae7e8)

**Reply by rtivital** [reply]
https://mantine.dev/styles/style-props/
